server/controllers/request.js

```diff
@@ -0,0 +1,99 @@
+const Request = require("../models/Request");
+const asyncHandler = require("express-async-handler");
+
+/*
+@route GET /requests/getRequests
+@desc for fetching all the requests for a particular user
+@access private
+*/
+exports.getRequests = asyncHandler( async( req, res, next ) => {
+    const  user_id  = req.query.user_id;
+    
+    const user = await Request.find({ user_id: user_id }, function(err) {
+        if(err) {
+            res.status(404).send("User not found!!");
+        }
+    })
+
+    if(user) {
+        res.status(200).json({
+            success: {
+                data: user,
+                message: "Requests fetched successfully"
+            }
+        })
+    }
+})
+
+/*
+@route POST /requests/postRequest
+@desc for adding a new request
+@access private
+*/
+exports.addRequest = asyncHandler( async( req, res, next) => {
+    const { user_id, sitter_id, start, end, status } = req.body;
+
+    const user = await Profile.findOne({_id: user_id}, function(err) {
+        if(err) {
+            res.status(404).send("User not found!!")
+        }
+    });
+    const sitter = await Profile.findOne({_id: sitter_id }, function(err) {
+        if(err) {
+            res.status(404).send("Sitter not found");
+        }
+    });
+
+    if(user && sitter) {
+        const request = await Request.create({
+            user_id,
+            sitter_id,
+            start,
+            end,
+            status
+        })
+
+        if(request) {
+            res.status(200).json({
```

Comment:
You can use status code for resource creation here :)

Ref: https://www.restapitutorial.com/httpstatuscodes.html